test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+			})
+		})
+
+		g.When("the service account is persistent-volume-binder", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:kube-system:persistent-volume-binder"}
+				tc.data.groups = [][]string{{"system:serviceaccounts:kube-system"}}
+				tc.data.namespaces = []string{"kube-system"}
+			})
+			g.It("should allow to update PersistentVolumeClaims", func() {
+				tc.data.resources = []string{"persistentvolumeclaims"}
+				tc.data.verbs = []string{"update"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			g.It("should allow to create PersistentVolumes", func() {
+				tc.data.resources = []string{"persistentvolumes"}
+				tc.data.verbs = []string{"create"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+
+		})
+
+		g.When("the service account is aws-cloud-provider", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:kube-system:aws-cloud-provider"}
+				tc.data.groups = [][]string{{"system:serviceaccounts:kube-system"}}
+			})
+			g.It("should allow to patch Nodes", func() {
+				tc.data.resources = []string{"nodes"}
+				tc.data.verbs = []string{"patch"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the service account is api-monitoring-controller", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:api-infrastructure:api-monitoring-controller"}
+			})
+			g.When("the namespace is kube-system", func() {
+				g.BeforeEach(func() {
+					tc.data.namespaces = []string{"kube-system"}
+				})
+				g.It("should allow to update 'skipper-default-filters' ConfigMap", func() {
+					tc.data.resources = []string{"configmaps"}
+					tc.data.verbs = []string{"update"}
+					tc.data.names = []string{"skipper-default-filters"}
+					tc.run(context.TODO(), cs, true)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+				g.It("should deny to update any other ConfigMap", func() {
+					tc.data.resources = []string{"configmaps"}
+					tc.data.verbs = []string{"update"}
+					// Technically, this should result in access undecided because we allow
+					// access to 'skipper-default-filters' ConfigMap only and we haven't
+					// specified a resource name in the test case.
+					// We consider access undecided cases also as denied.
+					tc.run(context.TODO(), cs, false)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+			})
+		})
+
+		g.When("the user is k8sapi_credentials-provider", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"zalando-iam:zalando:service:k8sapi_credentials-provider"}
+			})
+			g.It("should allow to get Secrets in kube-system namespace", func() {
```

Comment:
Writing secrets is again done in `admission-controller` 😠 Hopefully we find a better solution in the future.

Reply:
Actually RBAC also gives credentials provider read/write to secret. Maybe it makes sense to add a test for it as well?

Reply:
Yeah 🤞

Reply:
> Actually RBAC also gives credentials provider read/write to secret. Maybe it makes sense to add a test for it as well?

Looking at[ the `ClusterRoleBinding`](https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/manifests/roles/credentials-provider-rbac.yaml#L20-L27) attached to this user, it's allowed all operation on `Secrets` except `delete`. 

I created two test cases, one for deletion, another for "all other" operations: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/39e6e18d57eb00c4da9ef0a7d4a1fa4c044c91ee